deb/ar.go

```diff
@@ -94,15 +94,24 @@ func (d *Ar) Next() (*ArEntry, error) {
 	}
 
 	line := make([]byte, 60)
-
-	count, err := d.in.Read(line)
-	if err != nil {
-		return nil, err
+	pos := 0
+
+	var err error
+	for pos < 60 {
+		var count int
+		count, err = d.in.Read(line[pos:])
+		if (err == nil) || (err == io.EOF){
+			pos = pos + count
+		}
+		if (err == io.EOF) { break }
+		if (err != nil) { return nil, err }
 	}
-	if count == 1 && line[0] == '\n' {
-		return nil, io.EOF
+	if err == io.EOF {
+		if pos == 0 || (pos == 1 && line[0] == '\n') {
+			return nil, err
+		}
 	}
-	if count != 60 {
+	if pos < 60 {
```

Comment:
Can this code be implemented by using an `io.LimitReader` and `ioutils.ReadAll`?